doordeck-sdk/src/commonMain/kotlin/com/doordeck/multiplatform/sdk/internal/api/LockOperationsClient.kt

```diff
@@ -313,6 +317,46 @@ internal object LockOperationsClient : AbstractResourceImpl() {
         performOperation(baseOperationRequest, operationRequest)
     }
 
+    /**
+     * Batch share a lock
+     *
+     * @see <a href="https://developer.doordeck.com/docs/#batch-share-a-lock-v2">API Doc</a>
+     */
+    suspend fun batchShareLockRequest(batchShareLockOperation: LockOperations.BatchShareLockOperation) {
+        /**
+         * Verify whether the operation device currently supports the batch sharing operation
+         */
+        val isSupported = CapabilityCache.isSupported(batchShareLockOperation.baseOperation.lockId, CapabilityType.BATCH_SHARING_25)
+            ?: getSingleLockRequest(batchShareLockOperation.baseOperation.lockId).also {
+                CapabilityCache.put(batchShareLockOperation.baseOperation.lockId, it.settings.capabilities)
+            }.settings.capabilities.containsKey(CapabilityType.BATCH_SHARING_25)
+
+        /**
+         * If the device does not support the batch sharing operation, we will call the single-user sharing operation for each user individually
+         */
+        if (!isSupported) {
+            batchShareLockOperation.users.forEach { shareLock ->
+                // Recreate the base operation because we need to use a different JTI for each user
+                shareLockRequest(LockOperations.ShareLockOperation(batchShareLockOperation.baseOperation.withNewJti(), shareLock))
```

Comment:
Perhaps worth adding a custom exception for this method which returns a list of users who failed to be added?

Reply:
Done!